context/app/static/js/components/workspaces/AddDatasetsTable/hooks.ts

```diff
@@ -81,23 +79,26 @@ function useSearchAhead({ value, valuePrefix = '', uuidsToExclude = [] }: BuildI
 }
 
 function useDatasetsAutocomplete({
-  workspaceId,
+  workspaceDatasets = [],
   selectedDatasets = [],
   updateDatasetsFormState,
 }: {
-  workspaceId: number;
+  workspaceDatasets?: string[];
   selectedDatasets: string[];
   updateDatasetsFormState: (datasetUUIDS: string[]) => void;
 }) {
   const [inputValue, setInputValue] = useState('');
+  const [, setRefresh] = useState(false);
   const [autocompleteValue, setAutocompleteValue] = useState<SearchAheadHit | null>(null);
   const { toastSuccess } = useSnackbarActions();
 
   const removeDatasets = useCallback(
     (uuids: string[]) => {
-      const selectedDatasetsSet = new Set(selectedDatasets);
-      uuids.forEach((uuid) => selectedDatasetsSet.delete(uuid));
-      updateDatasetsFormState([...selectedDatasetsSet]);
+      const updatedDatasets = selectedDatasets.filter((uuid) => !uuids.includes(uuid));
+      updateDatasetsFormState(updatedDatasets);
+
+      // Trigger a re-render to update the table results
+      setRefresh((prev) => !prev);
```

Comment:
Without this, any attempts to delete datasets added via the search bar will not be visible until the page is interacted with and a re-render is triggered. This is likely the result of the dependency array in the useEffect in useCreateWorkspaceForm not including the list of datasets - however, if these are included, the component will get stuck in a re-rendering loop.